docs/OV_Runtime_UG/Python_API_exclusives.md

```diff
@@ -0,0 +1,143 @@
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [tensor_slice_mode]
+
+@endsphinxdirective
+
+## Running inference
+
+Python API supports extra calling methods to synchronous and asynchronous modes for inference.
+
+All infer methods allow users to pass data as popular numpy arrays, gathered in either Python dicts or lists.
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [passing_numpy_array]
+
+@endsphinxdirective
+
+Results from inference can be obtained in various ways:
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [getting_results]
+
+@endsphinxdirective
+
+### Synchronous mode - extended
+
+Python API provides different synchronous calls to infer model, which block the application execution. Additionally these calls return results of inference:
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [sync_infer]
+
+@endsphinxdirective
+
+### AsyncInferQueue
+
+Asynchronous mode pipelines can be supported with wrapper class called `AsyncInferQueue`. This class automatically spawns pool of `InferRequest` objects (also called "jobs") and provides synchronization mechanisms to control flow of the pipeline.
+
+Each job is distinguishable by unique `id`, which is in the range from 0 up to number of jobs specified in `AsyncInferQueue` constructor.
+
+Function call `start_async` is not required to be synchronized, it waits for any available job if queue is busy/overloaded. Every `AsyncInferQueue` code block should end with `wait_all` function. It provides "global" synchronization of all jobs in the pool and ensure that access to them is safe.
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [asyncinferqueue]
+
+@endsphinxdirective
+
+#### Acquire results from requests
+
+After the call to `wait_all`, jobs and their data can be safely accessed. Acquring of a specific job with `[id]` returns `InferRequest` object, which results in seamless retrieval of the output data.
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [asyncinferqueue_access]
+
+@endsphinxdirective
+
+#### Setting callbacks
+
+Another feature of `AsyncInferQueue` is ability of setting callbacks. When callback is set, any job that ends inference, calls upon Python function. Callback function must have two arguments. First is the request that calls the callback, it provides `InferRequest` API. Second one being called "userdata", passed values can be of any Python type and later used inside callback function.
```

Comment:
Second one,  called "userdata", provides (?) values, which can be of any Python type and later used inside callback function.

Reply:
Done